switchbot/devices/lock.py

```diff
@@ -1 +1,120 @@
+"""Library to handle connection with Switchbot Lock."""
 from __future__ import annotations
+
+import logging
+from typing import Any
+
+from cryptography.hazmat.primitives.ciphers import Cipher, algorithms, modes
+from bleak import BLEDevice
+
+from .device import SwitchbotDevice
+from ..const import LockStatus
+
+COMMAND_HEADER = "57"
+COMMAND_GET_CK_IV = f"{COMMAND_HEADER}0f2103"
+COMMAND_UNLOCK = f"{COMMAND_HEADER}0f4e01011080"
+COMMAND_LOCK = f"{COMMAND_HEADER}0f4e01011000"
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class SwitchbotLock(SwitchbotDevice):
+    """Representation of a Switchbot Lock."""
+
+    def __init__(self, device: BLEDevice, key_id: str, encryption_key: str, interface: int = 0, **kwargs: Any) -> None:
+        if len(key_id) == 0:
+            _LOGGER.error("key_id is missing")
+        elif len(key_id) != 2:
+            _LOGGER.error("invalid key_id")
+        if len(encryption_key) == 0:
+            _LOGGER.error("encryption_key is missing")
+        elif len(encryption_key) != 32:
+            _LOGGER.error("invalid encryption_key")
+        self._iv = None
+        self._cipher = None
+        self._key_id = key_id
+        self._encryption_key = bytearray.fromhex(encryption_key)
+        super().__init__(device, None, interface, **kwargs)
+
+    async def lock(self) -> bool:
+        """Send lock command."""
+        result = await self._send_command(COMMAND_LOCK)
+        return self._check_command_result(result, 0, {1})
+
+    async def unlock(self) -> bool:
```

Comment:
https://github.com/Danielhiversen/pySwitchbot/blob/a804c7863b253dfe0b09bdca3cd724d21c201492/switchbot/devices/curtain.py#L80

Probably should decorate this with `@update_after_operation` to make sure the locking worked